### PR TITLE
refactor for pytest 6 compatibility

### PR DESCRIPTION
**Pull request by syphar**
https://docs.pytest.org/en/stable/deprecations.html#node-construction-changed-to-node-from-parent

These are some changes regarding pytest 6 compatibility. We don't stay backwards compatible here, assuming people can upgrade to pytest >= 6

### CODE REVIEW

**Comment by syphar** [comment]
@Ashlett anything we can do here? Or should I reassign? 


This blocks the pytest 6 update, so it's kind of FF is it? :)

File: pytest_translations/po_files.py [hunk 1]
```diff
@@ -6,26 +6,26 @@
 
 
 class PoFile(File):
-    def __init__(self, path, parent):
-        super(PoFile, self).__init__(path, parent)
+    def __init__(self, fspath, parent):
+        super(PoFile, self).__init__(fspath, parent)
```
Comment (Ashlett) [inline pytest_translations/po_files.py]: if we're already simplifying `super` calls, let's do the same here:
```suggestion
        super().__init__(fspath, parent)
```

File: .gitignore [hunk 2]
```diff
@@ -35,20 +41,94 @@ pip-delete-this-directory.txt
...
+.mypy_cache/
+.dmypy.json
+dmypy.json
+
+# Pyre type checker
+.pyre/
+
+# pytype static type analyzer
+.pytype/
+
```
Comment (Ashlett) [inline .gitignore]: how is updating `.gitignore` related to pytest 6 compatibility?

**Review by Ashlett — CHANGES_REQUESTED** [review]
sorry for taking so long :pensive: I had a lot on my plate and this PR sort of fell by the wayside.

"We don't stay backwards compatible here, assuming people can upgrade to pytest >= 6" - not sure I understand: does that mean that if we update this package in the backend requirements (after this PR), we have to update pytest at the same time, otherwise our tests will fail? But the build in this PR is running on `pytest-5.4.3` and it passed, am I missing something?

File: .gitignore (hunk 2 above)
Comment (syphar) [reply .gitignore]: nope, I removed it and will do another PR

**Comment by syphar** [comment]
> sorry for taking so long 😔 I had a lot on my plate and this PR sort of fell by the wayside.

no worries :) it wasn't high prio


> "We don't stay backwards compatible here, assuming people can upgrade to pytest >= 6" - not sure I understand: does that mean that if we update this package in the backend requirements (after this PR), we have to update pytest at the same time, otherwise our tests will fail? But the build in this PR is running on `pytest-5.4.3` and it passed, am I missing something?

I'm not sure right now, let me recheck. Perhaps I'm missing something

**Comment by syphar** [comment]
@Ashlett I'm not sure why travis still used the old pytest version from some cache, but I just set `pytest>=6.1` in `requirements-dev.txt` to be sure. 

You're right, the new code also works with old pytest versions, though IMHO it's fine if we test only with the most recent version for this library

**Review by Ashlett — APPROVED** [review]
:+1: :sparkles:

**Pull request merged**